### DISCUSSION TITLE
Why are popover-related props passed to store and store provider

### DISCUSSION THREAD

**Discussion by ciampo** [ariakit/ariakit#4295] [Questions] [answered]
### Motivation

As I was reviewing the `Menu` API surface, I noticed that popover-related props (such as `placement`, `open`, ...) need to be passed to the store / store provider component, while all other popover-related props (`modal`, `gutter`, ...) are passed directly to the `Menu` component. The same is true for components with a similar structure, like `Select`.

I found this a bit counter-intuitive, especially given that the prop could be completely redundant in case the consumer of the component doesn't need a popover, and decides to render a `MenuList` (or `SelectList`) instead of a `Menu` (or a `SelectPopover`).

### Usage example

N/A

### Requirements

Is there a reason for this API design choice? Should we consider moving popover-related props to the components rendering the popover ?

**Comment by diegohaz** [comment, marked as answer]
In earlier versions, all those props were passed to the store. We changed this in [v0.2.0](https://github.com/ariakit/ariakit/blob/main/packages/ariakit-react/CHANGELOG-035.md#020).

The plan was to move `placement` too, but it wasn't technically feasible since other components rely on this state.

Regarding `open`, you can control this state by passing props to components (see [Ariakit Popover with open prop passed to component](https://stackblitz.com/edit/kpxfir?file=popover%2Fstyle.css,popover%2Findex.tsx&theme=dark)). However, this state ultimately belongs to the store, so internally, we'll sync this prop with the store state.

**Reply by ciampo** [reply]
Got it, thank you for the context!

> this state ultimately belongs to the store, so internally, we'll sync this prop with the store state.

Do you think this could be possible for `placement` too?

**Reply by diegohaz** [reply]
This might be possible, but I haven't tried it.

**Reply by ciampo** [reply]
> Regarding open, you can control this state by passing props to components (see [Ariakit Popover with open prop passed to component](https://stackblitz.com/edit/kpxfir?file=popover%2Fstyle.css,popover%2Findex.tsx&theme=dark)). However, this state ultimately belongs to the store, so internally, we'll sync this prop with the store state.

Just wanted to also flag that only the `open` prop is exposed, and that there are no equivalent props for the `defaultOpen` and the `setOpen` props, which means that it's still not really possible to pass those props to the menu popover component instead of the store/provider.
